bentoml/_internal/frameworks/onnxmlir.py

```diff
@@ -0,0 +1,396 @@
+from __future__ import annotations
+
+import os
+import sys
+import shutil
+import typing as t
+import logging
+import platform
+import subprocess
+from copy import deepcopy
+from typing import Iterable
+from typing import TYPE_CHECKING
+from pathlib import Path
+from contextlib import contextmanager
+
+import attr
+
+import bentoml
+from bentoml import Tag
+from bentoml.models import ModelContext
+from bentoml.models import ModelOptions
+from bentoml.exceptions import NotFound
+from bentoml.exceptions import InvalidArgument
+from bentoml.exceptions import BentoMLException
+from bentoml.exceptions import MissingDependencyException
+
+from ..utils.pkg import find_spec
+from ..utils.buildx import run_docker_cmd
+from ..utils.buildx import UNAME_M_TO_PLATFORM_MAPPING
+
+if TYPE_CHECKING:
+    from .. import external_typing as ext
+    from ..types import PathType
+    from ...types import ModelSignature
+    from ..models.model import ModelSignaturesType
+
+    class ExecutionSession:
+        def __init__(self, shared_lib_path: str, use_default_entry_point: bool = ...):
+            ...
+
+        def run(self, input: list[ext.NpNDArray]) -> list[ext.NpNDArray]:
+            ...
+
+        def input_signature(self) -> str:
+            ...
+
+        def output_signature(self) -> str:
+            ...
+
+        def entry_points(self) -> list[str]:
+            ...
+
+        def set_entry_point(self, name: str):
+            ...
+
+
+logger = logging.getLogger(__name__)
+
+MODULE_NAME = "bentoml.onnxmlir"
+MODEL_FILENAME = "saved_model.so"
+API_VERSION = "v1"
+
+# TODO: mount session to docker container and use prebuilt?
+DEV_MLIR_IMAGE = "onnxmlirczar/onnx-mlir-dev:latest"
+
+# release docker images to compile the binary
+RELEASE_MLIR_IMAGE = "onnxmlirczar/onnx-mlir:latest"
+DOCKER_WORKDIR = "/workdir"
+DOCKER_OUTDIR = "/output"
+COMPILE_ARG = ["--O3", "--EmitLib"]
```

Comment:
opportunity for user to customize this. This will be P1 item where we can allow end user to use other onnx-mlir options.